src/NuGet.Core/NuGet.Build.Tasks.Console/app.manifest

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+    <assemblyIdentity version="1.0.0.0" name="MyApplication.app"/>
```

Comment:
Will this cause any issues not having the correct assembly information?  Do we need any more settings like MSBuild.exe has?

https://github.com/dotnet/msbuild/blob/main/src/MSBuild/MSBuild.exe.manifest

Reply:
We probably don't need that at all.

We have a manifest for nuget.exe, maybe we should be matching that, https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.CommandLine/app.manifest.

Reply:
> Will this cause any issues not having the correct assembly information?

If you extract the app manifest of any .NET Framework app that does not specify its own app.manifest, then, as stated in the PR description, it gets roslyn's [default app manifest](https://github.com/dotnet/roslyn/blob/ee9808b0ec1d5b94dba1ade21319b61487588ced/src/Compilers/Core/Portable/Resources/default.win32manifest). The app manifest is unchanged from this default, including calling the assembly `MyApplication.app`.

> Do we need any more settings like MSBuild.exe has?

In Visual Studio, when you use the new item wizard to add an app manifest, it contains all the `supportedOS` compatibility, and possibly one or two other sections, but I question it's usefulness given that Roslyn's default doesn't contain it, and nobody has complained about compatibility issues yet. It also doesn't specify Windows 11. Does that mean none of these apps are compatible with Windows 11? 🤷

> We have a manifest for nuget.exe, maybe we should be matching that,

I considered that, but as hinted at in my previous paragraph, many of the settings are not in the default manifest when you build a .NET Framework app without a manifest. I don't know what all the other settings do, what unintended side effects they'll have, so I feel more comfortable using the default settings, and adding the one we know we want.

But if we as a team prefer to do things differently, we can.

Reply:
Following an internal discussion, we appear to be satisfied with the fact that it seems to work, despite the non-obvious name mismatch. FWIW, [the win32 assembly identity](https://learn.microsoft.com/en-us/windows/win32/sbscs/application-manifests#assemblyidentity) does not appear to have any relation to .NET assembly identities. It's unfortunate that they both call themselves assemblies, but they're not referring to the same thing.